packages/camera/camera_android/example/android/build.gradle

```diff
@@ -14,6 +14,19 @@ allprojects {
         google()
         mavenCentral()
     }
+    // Build the plugin project with warnings enabled. This is here rather than
+    // in the plugin itself to avoid breaking clients that have different
+    // warnings (e.g., deprecation warnings from a newer SDK than this project
+    // builds with).
+    gradle.projectsEvaluated {
+        project(":camera_android") {
+            tasks.withType(JavaCompile) {
```

Comment:
My gradle skill level is barely above "cargo cult", so if there's a better way to do this please let me know :)

Reply:
One improvement is to move this out of the `allprojects` block, so that it runs only once instead of once for each project in the build.

You can see the difference if you add `println(options.compilerArgs)` after the `options.compilerArgs << …` line. When this is outside the `allprojects` block (and the `options.compilerArgs << …` line is uncommented), that prints once per `JavaCompile` task of the plugin, with lines like:
```
[-Xlint:deprecation, -Xlint:unchecked, -Xlint:all, -Werror]
```

But when this is inside the `allprojects` block, that prints once per `JavaCompile` task of the plugin, per project of the whole build, and the last lines each look like:
```
[-Xlint:deprecation, -Xlint:unchecked, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror, -Xlint:all, -Werror]
```

I feel like there should be a way to simplify the `gradle.projectsEvaluated { project(":camera_android") { … } }`, too, but I'm not sure how. Trying `project(":camera_android").afterEvaluate { … }` doesn't seem to do anything.

Reply:
Thanks! That was cargo-culted; I just assumed I couldn't refer to a project defined elsewhere without it (and that it would therefore only actually have a match once) and so didn't investigate how exactly it worked. Fixed.